src/gretel_synthetics/generator.py

```diff
@@ -0,0 +1,236 @@
+    for ensuring reliable serializability without an excess amount of code tied to it.
+    """
+
+    config: "LocalConfig"
+    start_string: str = "<n>"
+    line_validator: Optional[Callable] = None
+    max_invalid: int = 1000
+
+    def serialize(self) -> bytes:
+        return cloudpickle.dumps(self)
+
+    @staticmethod
+    def deserialize(serialized: bytes) -> "Settings":
+        obj = cloudpickle.loads(serialized)
+        if not isinstance(obj, Settings):
+            raise TypeError("deserialized object is of type {}, not Settings".format(type(obj).__name__))
+        return obj
+
+
+@dataclass
+class gen_text:
+    """
+    A record that is yielded from the ``Generator.generate_next`` generator.
+
+    Attributes:
+        valid: True, False, or None. If the line passed a validation function,
+            then this will be ``True``. If the validation function raised an exception
+            then this will be automatically set to ``False``. If no validation function
+            is used, then this value will be ``None.``
+        text: The actual record as a string
+        explain: A string that describes why a record failed validation. This is the
+            string representation of the ``Exception`` that is thrown in a validation
+            function. This will only be set if validation fails, otherwise will be ``None.``
+        delimiter: If the generated text are column/field based records. This will hold the delimiter
+            used to separate the fields from each other.
+    """
+
+    valid: bool = None
+    text: str = None
+    explain: str = None
+    delimiter: str = None
+
+    def as_dict(self) -> dict:
+        """Serialize the generated record to a dictionary
+        """
+        return asdict(self)
+
+    def values_as_list(self) -> Optional[List[str]]:
+        """Attempt to split the generated text on the provided delimiter
+
+        Returns:
+            A list of values that are separated by the object's delimiter or None is there
+            is no delimiter in the text
+        """
+        if self.delimiter is not None:
+            tmp = self.text.rstrip(self.delimiter)
+            return tmp.split(self.delimiter)
+        return None
+
+
+class Generator:
+    """
+    A class for generating synthetic lines of text.
+
+    Text generation is initiated via the ``generate_next`` method, which returns an iterable that yields values
+    until the given number of _valid_ lines is returned. Each Generator also has a maximum budget of the number
+    of invalid lines that can be generated across _all_ calls to ``generate_next``.
+    """
+    settings: Settings
+    model: tf.keras.Sequential
+    sp: spm.SentencePieceProcessor
+    delim: str
+    total_invalid: int = 0
+    total_generated: int = 0
+
+    def __init__(self, settings: Settings):
+        """
+        Constructor.
+
+        Args:
+            settings: the generator settings to use.
+        """
```

Comment:
We are standardizing to put all class level and constructor doc strings into only the class level docstrings. By default Sphinx doesn't import the `__init__` doc strings. Also, this let's us standardize how we write class doc strings regardless of using `dataclass` or just vanilla classes.

Reply:
Done